pyproject.toml

```diff
@@ -31,30 +31,58 @@ skip_empty = true
 #fail_under = 80
 
 [tool.pyright]
-strict = ["starcraft"]
+strict = ["charmcraft"]
 pythonVersion = "3.8"
 pythonPlatform = "Linux"
 venvPath = ".tox"
 venv = "py38"
 
 [tool.mypy]
 python_version = "3.8"
+packages = [
+    "charmcraft",
+]
 exclude = [
     "build",
-    "tests",
     "results",
+    "tests",
+    "tools",
 ]
 warn_unused_configs = true
 warn_redundant_casts = true
 strict_equality = true
-strict_concatenate = true
-warn_return_any = true
+#warn_return_any = true
 disallow_subclassing_any = true
 disallow_untyped_decorators = true
-disallow_any_generics = true
+#disallow_any_generics = true
+
+[[tool.mypy.overrides]]
+# Ignore typing errors in most legacy packages until we fix them.
```

Comment:
do we have an issue to fix these?

Reply:
#1316

I also removed a couple of these now as changes since I created this PR fixed them.